rust/agents/relayer/src/settings/mod.rs

```diff
@@ -19,18 +20,29 @@ pub enum GasPaymentEnforcementPolicy {
     MeetsEstimatedCost {
         coingeckoapikey: Option<String>,
     },
+
+    /// The required amount of gas on the foreign chain has been paid according
+    /// to on-chain fee quoting.
+    OnChainFeeQuoting {
+        /// Optional fraction of gas which must be paid before attempting to run the transaction.
+        /// Must be written as `"numerator / denominator"` where both are integers.
+        #[serde(default = "default_gasfraction")]
+        gasfraction: String,
+    },
 }
 
 /// Config for gas payment enforcement
 #[derive(Debug, Clone, serde::Deserialize)]
 #[serde(tag = "type", rename_all = "camelCase")]
 pub struct GasPaymentEnforcementConfig {
     /// The gas payment enforcement policy
+    #[serde(flatten)]
     pub policy: GasPaymentEnforcementPolicy,
     /// An optional whitelist, where all matching messages will be considered
```

Comment:
Technically I did update the comment to say that if no whitelist was provided it would consider all as matching, but I do agree changing the name makes some sense here so I have done that.